### PR TITLE
Implements ogit-fetch

### PR DESCRIPTION
**Pull request by ulugbekna**
Implements `ogit-fetch` supports 

- various transport: git://, http(s), and SSH
- progress reporting
- support for 'src_ref:dst_ref' where src_ref value is written to dst_ref

### CODE REVIEW

File: src/git-unix/ogit-fetch/main.ml [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+  in
+  let ( >>?= ) = Lwt_result.bind in
+  let ( >>!= ) v f = Lwt_result.map_err f v in
+  let resolvers =
+    let git_scheme_resolver = Conduit_lwt.TCP.resolve ~port:9418 in
+    let ssh_cfg = ssh_cfg repository ssh_seed in
+    Cohttp_lwt_unix.Net.empty
+    |> Conduit_lwt.add Conduit_lwt.TCP.protocol git_scheme_resolver
+    |> Conduit_lwt.add ssh_protocol (ssh_resolve @@ Option.get ssh_cfg)
+    (* FIXME add support for SSH *)
```
Comment (dinosaure) [inline src/git-unix/ogit-fetch/main.ml:96]: This comment can be deleted now :+1:.

**Comment by dinosaure** [comment]
According the merge of #404, this PR should follow but it seems that:
1) the PR introduces some bad things such as `>>>> HEAD` (so, some unfinished resolution of conflicts).
2) `awa-ssh` is missing in OPAM files (you should add a `pin-depends` with https://github.com/mirage/awa-ssh/pull/19
   About this point, a release of Conduit will fix such requirement but it seems that the release is locked by some details

If you have the time, can you refresh this PR with the new state of `master` (I can try to do that if you want)? In other side, it's a great work - I mean your PRs (not only this one) discovered some weird cases and improve globally the code :+1: !

**Comment by ulugbekna** [comment]
Oops, sorry about the conflicts. I thought I rebased and pushed long time ago. 

I hope I got new dependencies for `git-unix` right. Is there a way (using opam) to check if all dune deps are present in the opam file?

**Comment by dinosaure** [comment]
> I hope I got new dependencies for git-unix right. Is there a way (using opam) to check if all dune deps are present in the opam file?

It exists a way to generate the OPAM file from the `dune` file, but I don't know how to use it :/

**Comment by ulugbekna** [comment]
I know about [this](https://dune.readthedocs.io/en/stable/opam.html#generating-opam-files) for generating opam files using dune, but that's not very helpful in detecting if I am missing some deps in my opam file which are actually included in the dune file.

File: git-unix.opam [hunk 2]
```diff
@@ -48,6 +48,6 @@ depends: [
 pin-depends: [
   [ "conduit.dev" "git+https://github.com/mirage/ocaml-conduit.git#8912d458b2f3e43245e99cf3cb74e9c00712a8b0" ]
   [ "conduit-lwt.dev" "git+https://github.com/mirage/ocaml-conduit.git#8912d458b2f3e43245e99cf3cb74e9c00712a8b0" ]
-  [ "awa" "git+https://github.com/mirage/awa-ssh.git#e54302ecd5fe541c2fbecb277cb11bba40032b81"]
-  [ "awa-conduit" "git+https://github.com/mirage/awa-ssh.git#e54302ecd5fe541c2fbecb277cb11bba40032b81"]
+  [ "awa.dev" "git+https://github.com/mirage/awa-ssh.git#e54302ecd5fe541c2fbecb277cb11bba40032b81"]
+  [ "awa-conduit.dev" "git+https://github.com/mirage/awa-ssh.git#e54302ecd5fe541c2fbecb277cb11bba40032b81"]
```
Comment (dinosaure) [inline git-unix.opam:52]: I think the commit is a part of `dinosaure/awa-ssh` instead of `mirage/awa-ssh`

**Comment by ulugbekna** [comment]
fixed and rebased :-)

**Pull request merged**